VultisigApp/VultisigApp/Services/CoinService/EvmCoinService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  EvmCoinService.swift
+//  VultisigApp
+//
+//  Created by Enrique Souza Soares on 05/07/24.
+//
+
+import Foundation
+
+@MainActor
+class EvmCoinService: CoinService {
+    
+    override func addDiscoveredTokens(nativeToken: Coin, to vault: Vault) async {
```

Comment:
Its better to make CoinService methods `async throws` and handle errors outside

Reply:
For discovering tokens there is not need to throw it. We can simply don't add the problematic item IMO.